WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsTimezoneDialog.java

```diff
@@ -0,0 +1,338 @@
+                }
+                return true;
+            }
+        });
+        mSearchView.setEnabled(false);
+        mSearchView.setIconifiedByDefault(false);
+
+        mEmptyView = view.findViewById(R.id.empty_view);
+        mProgressView = view.findViewById(R.id.progress_view);
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.Calypso_AlertDialog);
+        builder.setPositiveButton(android.R.string.ok, this);
+        builder.setNegativeButton(R.string.cancel, this);
+        builder.setView(view);
+
+        mSelectedTimezone = getArguments().getString(KEY_TIMEZONE);
+        requestTimezones();
+
+        return builder.create();
+    }
+
+    private void requestTimezones() {
+        Response.Listener<String> listener = new Response.Listener<String>() {
+            @Override
+            public void onResponse(String response) {
+                AppLog.d(AppLog.T.SETTINGS, "timezones requested");
+                if (isAdded()) {
+                    showProgressView(false);
+                    if (!TextUtils.isEmpty(response)) {
+                        loadTimezones(response);
+                    } else {
+                        AppLog.w(AppLog.T.SETTINGS, "empty response requesting timezones");
+                        dismissWithError();
+                    }
+                }
+            }
+        };
+
+        Response.ErrorListener errorListener = new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                AppLog.e(AppLog.T.SETTINGS, "Error requesting timezones", error);
+                if (isAdded()) {
+                    dismissWithError();
+                }
+            }
+        };
+
+        StringRequest request = new StringRequest(Constants.URL_TIMEZONE_ENDPOINT, listener, errorListener) {
+            @Override
+            protected Map<String, String> getParams() throws AuthFailureError {
+                return RestClientUtils.getRestLocaleParams(getActivity());
+            }
+        };
+
+        showProgressView(true);
+        RequestQueue queue = Volley.newRequestQueue(getActivity());
+        queue.add(request);
+    }
+
+    private void loadTimezones(@NonNull String responseJson) {
+        ArrayList<Timezone> timezones = new ArrayList<>();
+        try {
+            JSONObject jsonResponse = new JSONObject(responseJson);
+            JSONArray jsonTimezones = jsonResponse.getJSONArray("timezones");
+            for (int i = 0; i < jsonTimezones.length(); i++) {
+                JSONObject json = jsonTimezones.getJSONObject(i);
+                timezones.add(
+                        new Timezone(json.getString("label"), json.getString("value"))
+                );
+            }
+
+            mAdapter = new TimezoneAdapter(timezones);
+            mListView.setAdapter(mAdapter);
+            mSearchView.setEnabled(true);
+
+            new Handler().postDelayed(new Runnable() {
```

Comment:
Can we add a comment about why we need to do this? I think we also need to check for `isAdded` inside the Runnable but I am not 100% sure.